examples/rust/_examples_common/custom_header.rs

```diff
@@ -10,7 +10,10 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#[derive(Default, Debug)]
+use iceoryx2_bb_derive_macros::ZeroCopySend;
+
+#[derive(Default, Debug, ZeroCopySend)]
+#[type_name(CustomHeader)]
 #[repr(C)]
 pub struct CustomHeader {
```

Comment:
Not for this PR, but it would be nice if one could use something like `#[type_name(Self)]`

Reply:
Thinking further about it, this would prevent to name the type `Self`. Would that be a too big restriction? Alternatively, we could require strings in the attribute, which would solve the problem but then we would need to parse for invalid characters, like a space. On the other hand, does the C++ counterpart allow spaces?

So what do you think of
```rs
#[typename("CustomHeader")]
#[typename(Self)]
```

Reply:
@elBoberido I think we need to go for the string-literal approach otherwise one set a type name on the C++ side that can never be set on the Rust side and this features becomes mooo

Reply:
@elBoberido @elfenpiff I changed the implementation so that only string literals are allowed for the type name attribute. With that, type names set on the C++ side can also be set on the Rust side.
`#[type_name(Self)]` is currently not allowed but users can either provide the name with the attribute or use `core::any::type_name` by not specifying the attribute.